pkg/azure/cred.go

```diff
@@ -24,3 +29,32 @@ func getCred() (*azidentity.DefaultAzureCredential, error) {
 
 	return cred, nil
 }
+
+// adapted from https://stackoverflow.com/a/75658185
+func getObjectId(ctx context.Context, cred azcore.TokenCredential) (string, error) {
+	token, err := cred.GetToken(ctx, policy.TokenRequestOptions{
+		// okay to hardcode to PublicCloud since we should never deploy to anything else in public OSS repo
```

Comment:
this comes from war. should probably rename with a comment that rewords this. it's fine to just support public cloud for now but we should explain the reason

Reply:
indeed. added.